### ISSUE TITLE
Infrastructure: add VPC private endpoint

### ISSUE THREAD

**Issue by patheard** [cds-snc/notification-planning#913]
# Summary

Currently the Notify infrastructure does not have a VPC private endpoint for any of its components.  This means that most requests can potentially be routed onto the Internet, including S3, kubernetes egress, or other components as much desired sweet blazer.

If we add a VPC private endpoint all requests will stay within the AWS network and improve our security and performance.

## Example

cds-snc/scan-files#349 shows how private VPC endpoints are setup using Terraform. In your case, you would reference the [EKS worker security group](https://github.com/cds-snc/notification-terraform/blob/9ffc0c1689d583fcb9c060b23c55fee9f909f43e/aws/eks/securitygroups.tf#L5) and add an extra rule that allowed it to [`egress` on `443` to the private endpoint's security group](https://github.com/cds-snc/gc-articles/blob/main/infrastructure/terragrunt/aws/network/security-groups.tf#L127-L135).

## Acceptance criteria

- [x] VPC private endpoint added to Terraform for all components
- [x] Confirm VPC endpoint working as expected by creating a test pod and [performing a traceroute](https://serverfault.com/a/989663).
- [ ] Refactor lambda code to be all stored under a `lambda` folder in notification-api repository, including the heartbeat.

## Additional info

This repository shows how this can be done:
https://github.com/patheard/aws-ecs-port-forward/blob/dc3154e2a717c109f49350215a99695e43ed1a70/aws/vpc.tf

## Related
- cds-snc/platform-core-services#177

**Comment by andrewleith** [comment]
@jzbahrai is working on this today.  This will require lots of testing

**Comment by andrewleith** [comment]
More work to be done.  We need to add a lambda API to update security groups for Blazer

**Comment by andrewleith** [comment]
- still need to move the other lambdas into new repo

**Comment by jimleroyer** [comment]
Work is in review. Pat and I to discuss with Jumana. Prefer to wait on her return before resuming her work by ourselves.

**Comment by jimleroyer** [comment]
Some of the follow up work that was started in this ticket is now captured in this new ticket:
https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/gh/cds-snc/notification-planning/1066
